### ISSUE TITLE
Loading thumbnails is slow

### ISSUE THREAD

**Issue by pdehaan** [mozilla/chronicle#259]
Not sure if we're planning on adding the embed.ly crop and url2png tasks are going to be offloaded into a worker queue for pre-generation, but currently they cause the front end to load somewhat ... slowly.

![chronicle](https://cloud.githubusercontent.com/assets/557895/6030131/109bb0e4-abaa-11e4-86f7-8712ee4359c0.png)

It seems from my casual devtools browsing that it takes ~2.x seconds for each thumbnail generation (for cases that use url2png).

**Comment by nchapman** [comment]
Yeah ideally we'll precache these when the server handles the logic for which image to use.

**Comment by pdehaan** [comment]
Cool, cool, cool... Yeah, I was wondering if we'd need something similar for the favicons since some of those are HTTP and others are HTTPS, which is giving me some mixed protocol warnings. Not sure if we're planning on proxying all the favicons through embed.ly, or if we can just scrape them once and store them in Postgres as a datauri or something.

**Comment by nchapman** [comment]
> something similar for the favicons since some of those are HTTP and others are HTTPS

Yeah we do need to fix that... either through the display endpoint or the crop endpoint. Will you create an issue for this if we don't have one?

**Comment by pdehaan** [comment]
![](https://media.giphy.com/media/HjPbLbmep2aJO/giphy.gif)

Bug incoming!

**Comment by 6a68** [comment]
Thanks for filing, @pdehaan! Moving to backlog till we've got enough app to start optimizing it, somebody shout if we care about mixed-content warnings or slightly slow rendering right now
